src/Gizmos/planeRotationGizmo.ts

```diff
@@ -201,6 +269,100 @@ export class PlaneRotationGizmo extends Gizmo {
+        const fillRadians = 0;
+        const step = this.circleConstants.pi2 / this.circleConstants.tessellation;
+        for (let p = -Math.PI / 2; p < Math.PI / 2 - 1.5; p += step / 2) {
+            const path = [];
+            for (let i = 0; i < this.circleConstants.pi2; i += step) {
+                if (i < fillRadians) {
+                    const x = this.circleConstants.radius * Math.sin(i) * Math.cos(p);
+                    const z = this.circleConstants.radius * Math.cos(i) * Math.cos(p);
+                    const y = 0;
+                    path.push(new Vector3(x, y, z));
+                } else {
+                    path.push(new Vector3(0, 0, 0));
+                }
+            }
+
+            paths.push(path);
+        }
+
+        const mat = new StandardMaterial("", this.gizmoLayer.utilityLayerScene);
+        mat.diffuseColor = Color3.Yellow();
+        mat.backFaceCulling = false;
+        const mesh = Mesh.CreateRibbon("ignore", paths, false, false, 0, this.gizmoLayer.utilityLayerScene, true);
+        mesh.material = mat;
+        mesh.material.alpha = .25;
+        mesh.rotation.x = Math.PI / 2;
+        parentMesh.addChild(mesh);
+        return mesh;
+    }
+
+    private updateRotationPath(pathArr: any[], newFill: number): void {
+        // To update the Ribbon, you have to mutate the pathArray in-place
+        const step = this.circleConstants.pi2 / this.circleConstants.tessellation;
+        let tessellationCounter = 0;
+        for (let p = -Math.PI / 2; p < Math.PI / 2 - 1.5; p += step / 2) {
+            const path = pathArr[tessellationCounter];
+            if (path) {
+                let radianCounter = 0;
+                for (let i = 0; i < this.circleConstants.pi2; i += step) {
+                    if (path[radianCounter]) {
+                        if (i < Math.abs(newFill)) {
+                            const eie = (newFill > 0) ? i : i * -1;
+                            const pea = (newFill > 0) ? p : p * -1;
+                            path[radianCounter].x = this.circleConstants.radius * Math.sin(eie) * Math.cos(pea);
+                            path[radianCounter].z = this.circleConstants.radius * Math.cos(eie) * Math.cos(pea);
+                            path[radianCounter].y = 0;
+                        } else {
+                            path[radianCounter].x = 0;
+                            path[radianCounter].y = 0;
+                            path[radianCounter].z = 0;
+                        }
+                    }
+
+                    radianCounter++;
+                }
+            }
+
+            tessellationCounter ++;
+        }
+    }
+
+    private updateRotationCircle(mesh: Mesh, paths: any[], newFill: number, dragPlanePoint: Vector3): void {
+        this.updateRotationPath(paths, newFill);
+        mesh = Mesh.CreateRibbon("ribbon", paths, false, false, 0, this.gizmoLayer.utilityLayerScene, undefined, undefined, mesh);
```

Comment:
Will it be possible to update the mesh itself and not create a new one? The old mesh is passed here as a reference - I find changing the reference is very confusing for a developer looking at/using this function.
Also - if not possible - return the mesh, and make sure to dispose the old one.